### ISSUE TITLE
Give players the option of a male or female sprite

### ISSUE THREAD

**Issue by Qiangong2** [Tuxemon/Tuxemon#709]
This is actually a lot more involved than it sounds (at least in my opinion), because we need:

- Both sprites as options
- A small story at the beginning asking what gender you are (kind of like animal crossing)
- Animations for the female sprite
- Generalize the player sprite so it isn't hard-coded to use the male sprite

**Comment by bitcraft** [comment]
The player sprite can be changed right now by changing the "player_npc" value in the config.  Any existing sprite with walk animations would work fine.  As far as gender goes, I think that deserves its own issue.  I'd like the project to be inclusive, so that may require rethinking the options of male/female and the language around it, or at least prepare for alternative options.

**Comment by xirsoi** [comment]
I think the simplest would be to just offer sprites to choose from and ask for the preferred pronouns. Localisation can insert the pronoun where appropriate in dialogue (if it ever even comes up) and keep it as simple as that.

**Comment by bitcraft** [comment]
yeah, something like that seems fine to me

**Comment by Sanglorian** [comment]
That sounds ideal to me too. 

We currently have two options for the protagonist's sprite: 

https://wiki.tuxemon.org/index.php?title=Adventurer
https://wiki.tuxemon.org/index.php?title=Heroine

No reason why we couldn't have more, though! The only thing that distinguishes protagonist sprites from any other trainer sprite is that the protagonist also needs a back sprite.

**Comment by bitcraft** [comment]
hmmm, reminds me that we don't have a way to customize the back sprite.

**Comment by vnmabus** [comment]
I think in one campaign there was a female character attracted to the protagonist. Maybe that could be seen as undesirable by homosexual males or heterosexual females. I would rather have no romance whatsoever in a game like this one but I do not want to interfere in the writers' creative process.

**Comment by ultidonki** [comment]
I think there are 3 reasons you'd want to change a sprite:

1. Permanently, ie. write it to the tuxemon.cfg (or to a save file), as in this issue - change your sprite to be male or female.
2. Temporarily - ie. you might want to change the sprite to a 'swimming', or 'flying' one, and then change it back to what it was originally once that part of the gameplay is finished. 
3. Change the game object completely (ie. to another trainer, or tuxemon, as in #508, #509) 

Ignoring 3, we can do points 1 and 2 using an action. 
`change_player_sprite <npc_slug>,<permanent or blank>`
If you say 'permanent' as the 2nd parameter, then it should be written to the tuxemon.cfg (or, to the save file somewhere)
If you leave the <npc_slug> as blank, then it should go back to whatever is in the tuxemon.cfg (or load it from the save file)

These are just my messy thoughts? As it would be nice to get swimming looking better by having an action to change you to a swimmer while in the water. 

You may also need a condition: `is player_gender male`, depending on events. Or, you could just store it as a game variable. 

> I think in one campaign there was a female character attracted to the protagonist. 

I don't think there's any romance in any campaign yet, but you could store that as a game variable too, to trigger different events depending on what kind of romance you want.

**Comment by bitcraft** [comment]
i don't think that a "temporary" change makes sense for how the animations work.  when a game npc is assigned a sprite, that name of the sprite is used to get the animations for it.  currently, that is just standing or walking.  so changing the sprite for swimming, for example, would ideally be handled the same way we deal with walking, which is to look for frames like "red_<direction>_<animation>_<frame number>.png"....  or "red_swim_front_001.png".  So changing the sprite in the game would change all animations.

As for actually implementing swimming or something, thats a different subject, but wouldn't be a big deal and i think we would implement that in the engine rather than trying to make it work with map events.

**Comment by JaskRendix** [comment]
@Qiangong2 implemented, you can pick you start a new game and there is an action to change sprite.
